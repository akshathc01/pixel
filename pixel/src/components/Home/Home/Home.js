import React from 'react';
import GetCommunityImages from '../../GetCommunityImages/GetCommunityImages';
import Header from '../Header/Header';

const Home = () => {
    return (
        <div data-testid="test-Home" style={{ backgroundColor: "#12161f" }}>
            <Header />
            <GetCommunityImages />
        </div>
    );
};

export default Home;