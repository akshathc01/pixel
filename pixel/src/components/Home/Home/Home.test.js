import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Home from './Home'
import { BrowserRouter as Router } from 'react-router-dom';

test('test', () => {
    expect(true).toBe(true)
})

test('it will render Home component', () => {
    render(<Router><Home /></Router>)
    const homeElement = screen.getByTestId('test-Home')
    expect(homeElement).toBeInTheDocument()
})

test('matches snapshot for Home', () => {
    const tree = renderer.create(<Router><Home /></Router>).toJSON()
    expect(tree).toMatchSnapshot()
})