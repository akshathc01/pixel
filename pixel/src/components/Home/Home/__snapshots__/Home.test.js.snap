// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for Home 1`] = `
<div
  data-testid="test-Home"
  style={
    Object {
      "backgroundColor": "#12161f",
    }
  }
>
  <div
    data-testid="test-Header"
  >
    <div
      className="full-container"
      data-testid="test-Navbar"
    >
      <nav
        className="navbar navbar-expand-lg navbar-light p-0 bg-dark "
      >
        <div
          className="container-fluid text-light"
        >
          <a
            className="navbar-brand d-flex align-items-center"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="Icon"
              className="logo"
              src="icon.png"
              style={
                Object {
                  "width": "50px",
                }
              }
            />
            <h1
              className="pl-3 text-light"
            >
              Pixel
            </h1>
          </a>
          <button
            className="btn btn-danger ml-3"
          >
            <a
              className="text-light"
              href="/addImage"
              onClick={[Function]}
            >
               Upload Image 
            </a>
          </button>
          <button
            aria-controls="navbarNavAltMarkup"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarNavAltMarkup"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>
          <div
            className="collapse navbar-collapse justify-content-end"
            id="navbarNavAltMarkup"
          >
            <div
              className="navbar-nav"
            >
              <a
                className="nav-link active px-3 text-light h5  my-2"
                href="/"
                onClick={[Function]}
              >
                 Home 
                <span
                  className="sr-only"
                >
                  (current)
                </span>
                 
              </a>
              <a
                className="nav-link px-3 my-2"
              >
                <a
                  className="text-light h5 "
                  href="/dashboard"
                  onClick={[Function]}
                >
                  Images
                </a>
              </a>
              <button
                className="btn btn-outline-danger mx-3 my-2"
                type="button"
              >
                 
                <a
                  className="text-light"
                  href="/signedIn"
                  onClick={[Function]}
                >
                  Sign In
                </a>
                  
              </button>
              <button
                className="btn btn-danger mx-3 my-2"
              >
                <a
                  className="text-light"
                  href="/signedIn"
                  onClick={[Function]}
                >
                  Sign Up
                </a>
              </button>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </div>
  <section
    className="services-container"
    data-testid="test-getCommunityImages"
  >
    <div
      className="text-center"
    >
      <div
        className="form-group row d-flex justify-content-center pt-5"
      >
        <input
          className="form-control border border-danger col-5 "
          onChange={[Function]}
          placeholder="Type for finding images"
          type=""
        />
      </div>
    </div>
    <div
      className="d-flex justify-content-center align-items-center"
    >
      <div
        className="w-100 row pb-5  justify-content-center"
      />
    </div>
  </section>
</div>
`;
